### PR TITLE
ets:insert/2 insert order preserved when list with same keys for bag table

### PR DESCRIPTION
**Pull request by madlep**
Since OTP 23, when inserting a list of values into an ets `bag` table, values for the same key are inserted in reversed order (prior to OTP 22, they were inserted in the order in the list).

Implementing fix suggested by @sverker to build linked list of `DbTerm` values stored against a key by _appending_ (rather than _prepending_ as it had been doing).

Fix for https://github.com/erlang/otp/issues/6730 (see description there for code to reproduce + original correct behaviour on OTP 22, and regression behaviour on OTP >= 23)

### CODE REVIEW

File: erts/emulator/beam/erl_db_tree.c [hunk 1]
```diff
@@ -3533,11 +3533,11 @@ void* db_eterm_to_dbterm_tree_common(int compress, int keypos, Eterm obj)
     return term;
 }
 
-void* db_dbterm_list_prepend_tree_common(void *list, void *db_term)
+void* db_dbterm_list_append_tree_common(void *last_term, void *db_term)
```
Comment (madlep) [inline erts/emulator/beam/erl_db_tree.c:3536]: Not sure if/where this code is even used, but have modified it to be consistent with `erl_db_hash` for consistency.

Comment (sverker) [reply erts/emulator/beam/erl_db_tree.c:3536]: It's for `ordered_set` tables.

**Comment by sverker** [comment]
I pushed a commit with some documentation improvements.
@madlep @gomoripeti What do you think?

**Comment by madlep** [comment]
> I pushed a commit with some documentation improvements. @madlep @gomoripeti What do you think?

makes sense to me 👍

**Comment by sverker** [comment]
Expanded the test case to include provoking bug for `duplicate_bag`.
Squashed into two commits and merged.

@madlep Thanks for good report and fix.

**Pull request merged**